tools/bcr_validation.py

```diff
@@ -225,6 +225,15 @@ def verify_module_dot_bazel(self, module_name, version):
             if actual_integrity != expected_integrity:
               self.report(BcrValidationResult.FAILED, f"The patch file `{patch_file}` has expected integrity value `{expected_integrity}`, but the real integrity value is `{actual_integrity}`.")
             apply_patch(source_root, source["patch_strip"], str(patch_file.resolve()))
+    if "overlay" in source:
+      version_dir = self.registry.get_version_dir(module_name, version)
+      for overlay_file, expected_integrity in source["overlay"].items():
+        overlay_src = version_dir / overlay_file
+        overlay_dst = source_root / overlay_file
+        actual_integrity = integrity(read(overlay_src))
+        if actual_integrity != expected_integrity:
+          self.report(BcrValidationResult.FAILED, f"The overlay file `{overlay_file}` has expected integrity value `{expected_integrity}`, but the real integrity value is `{actual_integrity}`.")
+        shutil.copy2(overlay_src, overlay_dst)
```

Comment:
Not on CI, the Linux VM running this check is thrown away after each build. But it may be a problem if you run this locally.

I guess we can add a check to verify `overlay_file` is a relative path, then overlay_src must be under version_dir and overlay_dst must be under source_root?

Reply:
If we also check that the relative path doesn't have `..` segments, then yes. Maybe there is a Python function that does this for us? Symlinks would make this difficult, but as a best-effort check, that could work.

Reply:
added a check for this